google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -471,3 +471,147 @@ def list(
+                form:
+                ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``.
+                The key needs to be in the same region as where the compute resource is created.
+                If this is set, then all
+                resources created by the PipelineJob will
+                be encrypted with the provided encryption key.
+                If not specified, encryption_spec of original PipelineJob will be used.
+            labels (Dict[str,str]):
+                Optional. The user defined metadata to organize PipelineJob.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to create this PipelineJob.
+                Overrides credentials set in aiplatform.init.
+            project (str),
+                Optional. The project that you want to run this PipelineJob in.
+                If not set, the project set in original PipelineJob will be used.
+            location (str),
+                Optional. Location to create PipelineJob.
+                If not set, location set in original PipelineJob will be used.
+
+        Returns:
+            A Vertex AI PipelineJob.
+
+        Raises:
+            ValueError: If job_id or labels have incorrect format.
+        """
+        ## Initialize an empty PipelineJob
+        if not project:
+            project = self.project
+        if not location:
+            location = self.location
+        if not credentials:
+            credentials = self.credentials
+
+        cloned = self.__class__._empty_constructor(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+        cloned._parent = initializer.global_config.common_location_path(
+            project=project, location=location
+        )
+
+        ## Get gca_resource from original PipelineJob
+        pipeline_job = json_format.MessageToDict(self._gca_resource._pb)
+
+        ## Set pipeline_spec
+        pipeline_spec = pipeline_job["pipelineSpec"]
+        if "deploymentConfig" in pipeline_spec:
+            del pipeline_spec["deploymentConfig"]
+
+        ## Set caching
+        if enable_caching is not None:
+            _set_enable_caching_value(pipeline_spec, enable_caching)
+
+        ## Set job_id
+        pipeline_name = pipeline_spec["pipelineInfo"]["name"]
+        cloned.job_id = job_id or "cloned-{pipeline_name}-{timestamp}".format(
+            pipeline_name=re.sub("[^-0-9a-z]+", "-", pipeline_name.lower())
+            .lstrip("-")
+            .rstrip("-"),
+            timestamp=_get_current_time().strftime("%Y%m%d%H%M%S"),
+        )
+        if not _VALID_NAME_PATTERN.match(cloned.job_id):
+            raise ValueError(
+                "Generated job ID: {} is illegal as a Vertex pipelines job ID. "
+                "Expecting an ID following the regex pattern "
+                '"[a-z][-a-z0-9]{{0,127}}"'.format(cloned.job_id)
```

Comment:
Consider interpolating `_VALID_NAME_PATTERN` in the `ValueError` message using `_VALID_NAME_PATTERN.pattern` rather than duplicating the string, that way they do not need to be kept in sync